petclinic-frontend/src/features/inventories/AddInventoryForm.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect } from 'react';
+import { Inventory } from './models/Inventory';
+import { getAllInventoryTypes } from '@/features/inventories/api/getAllInventoryTypes.ts';
+import addInventory from '@/features/inventories/api/addInventory.ts';
+import { InventoryType } from '@/features/inventories/models/InventoryType.ts';
+import './AddInventoryForm.css';
+
+interface AddInventoryProps {
+  show: boolean;
+  handleClose: () => void;
+  refreshInventoryTypes: () => void;
+}
+
+const AddInventoryForm: React.FC<AddInventoryProps> = ({
+  show,
+  handleClose,
+  refreshInventoryTypes,
+}: AddInventoryProps): React.ReactElement | null => {
+  const [inventoryName, setInventoryName] = useState<string>('');
+  const [inventoryType, setInventoryType] = useState<string>('');
+  const [inventoryDescription, setInventoryDescription] = useState<string>('');
+  const [inventoryTypes, setInventoryTypes] = useState<InventoryType[]>([]);
+
+  useEffect(() => {
+    async function fetchInventoryTypes(): Promise<void> {
+      try {
+        const types = await getAllInventoryTypes();
+        setInventoryTypes(types);
+      } catch (error) {
+        console.error('Error fetching inventory types:', error);
+      }
+    }
+    fetchInventoryTypes();
+  }, []);
+
+  // Handling form submission
+  const handleSubmit = async (e: React.FormEvent): Promise<void> => {
+    e.preventDefault();
+    const selectedInventoryType = inventoryTypes.find(
+      type => type.type === inventoryType
+    );
+
+    if (!selectedInventoryType) {
+      console.error('Invalid inventory type selected.');
+      return;
+    }
+
+    const newInventory: Omit<Inventory, 'inventoryId'> = {
+      inventoryName,
+      inventoryType: selectedInventoryType.type,
+      inventoryDescription,
+    };
+
+    try {
+      await addInventory(newInventory as Omit<Inventory, 'typeId'>);
```

Comment:
You're also using typeId here. I suggest changing it to inventoryId like in the addInventory page.